### PR TITLE
Add release process for Puppet.

### PR DESCRIPTION
**Pull request by ndmckinley**
This adds a release (minor / patch version releases) process for Puppet.  It's been tested with the Logging module, which was released without trouble.

-----------------------------------------------------------------
# [all]
CI changes only.

### CODE REVIEW

File: .ci/containers/puppet-forge/in.rb [hunk 1]
```diff
@@ -27,6 +25,7 @@
 release_tarball = release_slug + '.tar.gz'
 release.download(Pathname(release_tarball))
 release.verify(Pathname(release_tarball))
-PuppetForge::Unpacker.unpack(release_tarball, ARGV[0], '/tmp/resource_download')
+PuppetForge::Unpacker.unpack(release_tarball, '/tmp/resource_unpack', '/tmp/resource_download')
+FileUtils.cp_r('/tmp/resource_unpack/.', ARGV[0])
```
Comment (SirGitsalot) [inline .ci/containers/puppet-forge/in.rb]: How big is the tarball? Do we need to clean up here to help avoid full disks?

File: .ci/ci.yml.tmpl [hunk 2]
```diff
@@ -113,8 +119,28 @@ resources:
         uri: git@github.com:((github-account.username))/puppet-google-{{module}}.git
 {% endif %}
         private_key: ((repo-key.private_key))
+
+{% if module != "bigquery" %}
```
Comment (SirGitsalot) [inline .ci/ci.yml.tmpl]: What's special about bigquery? Presumably that it's not done yet; maybe add a TODO or file a bug to remember to enable it here and below once it's done.

Comment (SirGitsalot) [reply .ci/ci.yml.tmpl]: Or should there be a different means of excluding in-progress products, so we don't accidentally publish incomplete future products? (asked by someone who knows nothing about Puppet Forge)

Comment (ndmckinley) [reply .ci/ci.yml.tmpl]: Very much in agreement.  Added that, also refactored the whole deal so it's separate files and is deployed to separate pipelines.  That will be a big help for keeping all this manageable.

**Pull request merged**